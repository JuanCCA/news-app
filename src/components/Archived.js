import React from 'react'
import PropTypes from 'prop-types'

const Archived = () => {
    return (
        <>
           <h1>Archivados</h1> 
        </>
    )
}

Archived.propTypes = {

}

export default Archived
