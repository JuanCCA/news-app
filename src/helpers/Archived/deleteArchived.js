import { fetchAnonymous } from "../Petitions/fetch";

export const deleteArchived = async (_id) => {

    const payload = {
        _id: _id
    }

    
    const resp = await fetchAnonymous('archived/delete_archived', payload, 'DELETE')
    const response = await resp.json()
    return response
}

