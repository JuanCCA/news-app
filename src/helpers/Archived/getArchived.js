import {fetchAnonymous} from '../Petitions/fetch'

export const getArchived = async () => {

    const resp = await fetchAnonymous('archived/get_archived', null)
    const archived = await resp.json()

    return archived

}