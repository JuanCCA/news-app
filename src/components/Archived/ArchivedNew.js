import React from 'react'
import Card from 'react-bootstrap/Card'
import {deleteArchived} from '../../helpers/Archived/deleteArchived'
import { formatDate } from '../../helpers/Utils/dateFormater'
import Button from 'react-bootstrap/Button'


const ArchivedNew = ({_id,title,description,date,content,author,archiveDate,setArchived}) => {
    
    
    const formatedDate = formatDate(date)
    // Delete an archived new from DB
    const handlerDeleteArchived = async() => {
        const resp = await deleteArchived(_id)
        if(resp === 200){
            handlerDeleteFromList()
        }
    }

    // Delete this item from the list 
    const handlerDeleteFromList = () => {
        setArchived( (news) => {
            console.log(news)
            return news.filter( doc => doc._id !== _id)
        })
    }


    return (
        <Card style={{ width: '90%' }} className="my-3">
        <Card.Body>
            <Card.Title>{title}</Card.Title>
            <Card.Subtitle className="mb-2 text-muted">{description}</Card.Subtitle>
            <Card.Text>{content}</Card.Text>
            <Card.Footer >

            <div className="row">
                <div className="col">
                    Archived date: {formatedDate} 
                </div>
                <div className="col">
                    Author: {author}
                </div>
                <div className="col">
                    <Button onClick={handlerDeleteArchived}
                            variant="dark">
                        <img 
                            src="./delete.png"
                            width="25"
                            height="25"
                            className="d-inline-block align-top"   
                            alt=""                 
                            >
                        </img>
                        Delete
                    </Button>
                </div>
            </div>
                
            </Card.Footer>
        </Card.Body>
        </Card>
    )
}

export default ArchivedNew
