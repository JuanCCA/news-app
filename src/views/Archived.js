import React, {useState} from 'react'
import ArchivedNew from '../components/Archived/ArchivedNew'
import {useFetchArchived} from '../hooks/useFetchArchived'

const Archived = () => {
    
    const [archived,setArchived] = useState([])
    

    const { data:archivedList, loading } = useFetchArchived( archived );



    return (
        <>
            <div className="d-flex justify-content-center">
                <p className="display-1 m-auto">Archived</p>
            </div>
            <div align="center">
                <ol>
                { loading && <p className="animate__animated animate__flash">Loading</p> }
                    {
                        archivedList.map( item => (
                            <ArchivedNew key={item._id} {...item} setArchived={setArchived} />
                        ))
                    }
                </ol>
            </div>
            
        </>
    )
}



export default Archived
