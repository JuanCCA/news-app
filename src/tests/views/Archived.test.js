import React from 'react';
import '@testing-library/jest-dom';

import { shallow } from 'enzyme';
import  Archived  from '../../views/Archived';




describe('Testing <Archived /> Component', () => {
    

    let useFetchArchived = jest.fn()
    let wrapper = shallow(<Archived />);

    beforeEach( () => {
        jest.clearAllMocks();
        wrapper = shallow(<Archived />);
    });

    test('should create the component correctly', () => {
        
        expect( wrapper ).toMatchSnapshot();

    });

    test('should call useFetchArchived', () => {
  

        
    })
    




})

