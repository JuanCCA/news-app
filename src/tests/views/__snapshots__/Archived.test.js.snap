// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Archived /> Component should create the component correctly 1`] = `
<Fragment>
  <div
    className="d-flex justify-content-center"
  >
    <p
      className="display-1 m-auto"
    >
      Archived
    </p>
  </div>
  <div
    align="center"
  >
    <ol>
      <p
        className="animate__animated animate__flash"
      >
        Loading
      </p>
    </ol>
  </div>
</Fragment>
`;
